src/utils/uemParser.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+export class UEMParser {
+    public static findSObjects(json: Object): Array<string> {
+        const sObjects = UEMParser.findObjectsWithValues(json, [
+            'mcf/list',
+            'mcf/timedList',
+            'mcf/genericLists'
+        ]);
+        const results: string[] = [];
+
+        sObjects.forEach((obj) => {
+            let properties = obj['properties' as keyof Object];
+            let objectApiName = properties[
+                'objectApiName' as keyof Object
+            ] as unknown as string;
+
+            // Only include unique values in the array.
+            if (!results.includes(objectApiName)) {
+                results.push(objectApiName);
+            }
+        });
+
+        return results;
+    }
+
+    static findObjectsWithValues(
```

Comment:
Sorry, I'm still not convinced on "searching" the JSON. The JSON in question in this case is a well-known data structure. We should be querying it, not searching it. For all of its unwieldiness, it's still deterministic that `objectApiName` keys can be found at `view.regions.components.components[_x_].regions.components.components[_y_].properties.objectApiName` (or something along those lines). I haven't looked across all of the templates to see whether or not there may be a similar path to other ones—but if there is, it will be just as deterministic.

I feel like we should be deliberate about access these data properties. To me, searching for them is just opening a door to false positives, with no real upside. We should query the data structure we know.

Reply:
I wasn't sure if it will always be in that location. I suggested to @sfdctaka that as long as we find an `mcf/list` or `mcf/timedList`, then from there `.properties.objectApiName` should be populated. I thought that with the UEM file, it can be arbitrarily deep and nested in other components such as an `mcf/card` or something else perhaps.

Reply:
I support continuing to have some measure of a UEM "fronting" class for these data structures, because they are so unwieldy, and accessing properties therein in a more friendly encapsulated manner could be useful down the road.

Reply:
> I thought that with the UEM file, it can be arbitrarily deep and nested in other components such as an `mcf/card` or something else perhaps.

It'd be good to understand if that's the case, and then if so what the constraints are on that definition, and how we'd respond to them.

My point is more that we should strive to leverage our understanding of this data structure that we're supporting. To do a "find" is kind of punting on that understanding, and to me exposes a gap that we need to close.

Reply:
cc: @Taost

Reply:
I wrote a xpath-ish query function to replace the recursive deep search. @khawkins please check it out.